avocado/core/suite.py

```diff
@@ -0,0 +1,141 @@
+from enum import Enum
+from uuid import uuid1
+
+from .exceptions import OptionValidationError
+from .future.settings import settings
+from .loader import loader, LoaderError, LoaderUnhandledReferenceError
+from .resolver import resolve
+from .tags import filter_test_tags
+from .test import DryRunTest
+from .utils import resolutions_to_tasks
+from .varianter import Varianter
+
+
+class TestSuiteError(Exception):
+    pass
+
+
+class TestSuiteStatus(Enum):
+    RESOLUTION_NOT_STARTED = 0
+    TESTS_NOT_FOUND = 1
+    TESTS_FOUND = 2
+    UNKNOWN = 3
+
+
+class TestSuite:
+    def __init__(self, name, config, tests=None):
+        self.name = name
+        self.tests = tests or None  # this will force None when tests = []
+
+        # Create a complete config dict with all registered options + custom
+        # config
+        self.config = settings.as_dict()
+        if config:
+            self.config.update(config)
+
+        self._variant = None
+        self._references = None
+
+        if config.get('run.dry_run.enabled'):
+            self.convert_to_dry_run()
```

Comment:
It's expected that some Job API users will interact with `TestSuite`, so I believe this method should not be public at this time, given that it's limited to the loader implementation.  Also, it's being called indiscriminately here, so it causes a crash if the resolver implementation is used.  Sample job:

```python
#!/usr/bin/env python3

import sys
from avocado.core.job import Job

config = {'run.references': ['examples/tests/passtest.py:PassTest.test'],
          'run.test_runner': 'nrunner',
          'run.dry_run.enabled': True}

with Job(config) as j:
    sys.exit(j.run())
```

```
$ ./examples/jobs/passjob.py 

Avocado crashed: TypeError: 'Task' object is not subscriptable
Traceback (most recent call last):

  File "/home/cleber/src/avocado/avocado/avocado/core/job.py", line 455, in create_test_suite
    runner=self._test_runner_name)

  File "/home/cleber/src/avocado/avocado/avocado/core/suite.py", line 103, in from_config
    return cls.from_config_with_resolver(config, name)

  File "/home/cleber/src/avocado/avocado/avocado/core/suite.py", line 141, in from_config_with_resolver
    tests=tasks)

  File "/home/cleber/src/avocado/avocado/avocado/core/suite.py", line 40, in __init__
    self.convert_to_dry_run()

  File "/home/cleber/src/avocado/avocado/avocado/core/suite.py", line 92, in convert_to_dry_run
    self.tests[i] = [DryRunTest, self.tests[i][1]]

TypeError: 'Task' object is not subscriptable

Please include the traceback info and command line used on your bug report
Report bugs visiting https://github.com/avocado-framework/avocado/issues/new
```